containers/proxy-tftpd-image/tftp_wrapper.py

```diff
@@ -55,20 +58,90 @@ def read(self, requested):
            self._content = b''
         return data
     def size(self):
-        return int(self._request.headers['content-length'])
+        return self._size
     def close(self):
         self._request.close()
 
+class HttpResponseDataFiltered(HttpResponseData):
+    def __init__(self, url, capath, fqdn):
+        # request file by url and store it
+        self._fqdn = fqdn
+        self._request = requests.get(url, stream=True, verify=capath)
+        if self._request.status_code == 404:
+            raise FileNotFoundError()
+        self._request.raise_for_status()
+
+        self._stream = None
+        self._content = self._request.content
+        self.contentFilter()
+        self._size = len(self._content)
+
+    def contentFilter(self):
+        raise NotImplementedError()
+
+
+class HttpResponseDataFilteredPXE(HttpResponseDataFiltered):
+    def contentFilter(self):
+        new_content = ''
+        have_entry = False
+        entry = ''
+        in_entry = False
+        for line in self._content.decode('utf-8').splitlines():
+            if in_entry:
+                 if line.startswith(' ') or line.startswith('\t'):
+                     entry += line + '\n'
+                 else:
+                    in_entry = False
+                    if " MASTER=" + self._fqdn in entry:
+                        have_entry = True
+                        new_content += entry
+                    entry = ''
+            if not in_entry:
+                if line.startswith('LABEL'):
+                    entry = line + '\n'
+                    in_entry = True
+                else:
+                    if not line.startswith('ONTIMEOUT'): #ONTIMEOUT points to deleted entry
+                        new_content += line + '\n'
+        if not have_entry:
+            return # there are no specific entries, keep the content unchanged
+        self._content = new_content.encode('utf-8')
+
+class HttpResponseDataFilteredGrub(HttpResponseDataFiltered):
+    def contentFilter(self):
+        new_content = ''
+        have_entry = False
+        entry = ''
+        in_entry = False
+        for line in self._content.decode('utf-8').splitlines():
+            if in_entry:
+                 entry += line + '\n'
+                 if line.rstrip().endswith('}'):
+                    in_entry = False
+                    if " MASTER=" + self._fqdn in entry:
```

Comment:
This looks like a generic kernel option. There is nothing preventing a SUMA admin from adding this manually. Thus leading to "swallowed" bootloader entries.

Reply:
This is true. Maybe we can also try to find `MINION_ID_PREFIX` option and only if both are present, then filter it? To limit accidental overreach outside of retail scenario.

Reply:
Hmm.. what kernel options documentation? 😅

Reply:
I believe that this is a possible candidate for adding this piece of information:

https://www.uyuni-project.org/uyuni-docs/en/uyuni/specialized-guides/salt/salt-formula-pxe.html#retail.sect.formulas.pxe.kernelparams

Reply:
PXE formula is unrelated to Cobbler, so this is not a good place.

This is better: https://github.com/uyuni-project/uyuni-docs/blob/manager-4.3/modules/retail/pages/containerized-saltboot.adoc

Anyway, this is documentation for Retail. In non-retail environment, MASTER= does nothing, it is quite unlikely that someone uses it by accident, so I think it is safe.

Reply:
Although not happy, I will agree with the fact that it is unlikely that this is added manually.